openerp/addons/base/res/res_partner.py

```diff
@@ -387,6 +393,14 @@ def onchange_state(self, state_id):
             return {'value': {'country_id': state.country_id.id}}
         return {'value': {}}
 
+    @api.onchange('image')
+    def onchange_image(self):
```

Comment:
I think this method will crash if image is set (gratavar not defined). Simplify the condition

if not self.image and self.email: self.image = blahblah else do nothing